usaspending_api/common/etl/spark.py

```diff
@@ -560,3 +564,179 @@ def create_ref_temp_views(spark: SparkSession, create_broker_views: bool = False
+    logger=None,
+    file_format="csv",
+) -> List[str]:
+    """PySpark impl of Hadoop 2.x copyMerge() (deprecated in Hadoop 3.x)
+    Merges files from a provided input directory and then redivides them
+        into multiple files based on a provided maximum row count parameter.
+    Args:
+        spark: passed-in active SparkSession
+        parts_dir: Path to the dir that contains the input parts files. The parts dir name
+            determines the name of the merged files.
+        header: A comma-separated list of field names, to be placed as the first row of every final CSV file.
+            Individual part files must NOT therefore be created with their own header.
+        max_rows_per_merged_file: Final CSV data will be subdivided into numbered files so that there is not more than
+            this many rows in any file written. Only if the total data exceeds this value will multiple files be
+                created with a pattern of ``{merged_file}_N.{extension}`` with N starting at 1.
+        logger: The logger to use. If one note provided (e.g. to log to console or stdout) the underlying JVM-based
+            Logger will be extracted from the ``spark`` ``SparkSession`` and used as the logger.
+        file_format: The format of the part files and the format of the final merged file, e.g. "csv"
+
+    Returns:
+        A list of file paths where each element in the list denotes a path to
+            a merged file that was generated during the copy merge.
+    """
+    overwrite = True
+    if not logger:
+        logger = get_jvm_logger(spark)
+    hadoop = spark.sparkContext._jvm.org.apache.hadoop
+    conf = spark.sparkContext._jsc.hadoopConfiguration()
+    parts_dir_path = hadoop.fs.Path(parts_dir)
+
+    # See comment in docstring about how FileSystem is determined
+    fs = parts_dir_path.getFileSystem(conf)
+
+    if not fs.exists(parts_dir_path):
+        raise ValueError("Source directory {} does not exist".format(parts_dir))
+
+    file = parts_dir
+    file_path = hadoop.fs.Path(file)
+
+    # Don't delete first if disallowing overwrite.
+    if not overwrite and fs.exists(file_path):
+        raise Py4JError(
+            spark._jvm.org.apache.hadoop.fs.FileAlreadyExistsException(f"{str(file_path)} " f"already exists")
+        )
+    part_files = []
+
+    for f in fs.listStatus(parts_dir_path):
+        if f.isFile():
+            file_path = f.getPath()
+            if file_path.getName().startswith("_"):
+                logger.debug(f"Skipping non-part file: {file_path.getName()}")
+                continue
+            logger.debug(f"Including part file: {file_path.getName()}")
+            part_files.append(f.getPath())
+    if not part_files:
+        logger.warn("Source directory is empty with no part files. Attempting creation of file with CSV header only")
+        out_stream = None
+        try:
+            out_stream = fs.create(hadoop.fs.Path(parts_dir), overwrite)
```

Comment:
> Not sure if this makes sense to write the header-only .csv file to the location of the `parts_dir`. The `parts_dir` would be a prefix in S3 (like a folder in the FileSystem), and in this case an empty one (or at least empty of any part files, it could have a `_SUCCESS` file).
> 
> Below, when there are part files to merge (combine) the write via `fs.create` is to `partial_merged_file_path`, which is then renamed to `merged_file_path`. So probably `merged_file_path` should be used here instead of `parts_dir`.
> 
> Which means you need to declare that `merged_file_name` and `merged_file_path` vars earlier so they can be used in this case of empty parts dir.

This behavior is how you implemented the method. I assumed you had a reason and did it as it should be.

Reply:
Fixed